### PR TITLE
Privacy experience form updates

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/3320

Frontend buddy to https://github.com/ethyca/fides/pull/3387

### Code Changes

* [x] Updates the privacy experience config forms based off of [the new design](https://www.figma.com/file/dmEwdK3xZwjKfGVQ9t66xe/Fides-v.2-Master-Working-Files?type=design&node-id=6980-351571&t=VJkPvKvHJpxtzkeQ-0)
* [x] Updates cypress tests

### Steps to Confirm

* Run fidesplus with our fancy new feature flag system! I've already tagged `fides`, so you should be able to use `nox -s "build(slim)" -- dev_prerelease core_version=2.14.1a1` followed by `nox -s "dev(slim)" `
* Run the admin-ui in `clients` via `npm run dev-admin-ui`
* POST some experience configs, at least one overlay and one privacy center type. I used this JSON as a starting point
```json
{
  "title": "I am a banner",
  "description": "Do it now!",
  "component": "overlay",
  "regions": [],
  "acknowledge_button_label": "ok",
  "accept_button_label": "sounds good",
  "reject_button_label": "nah",
  "link_label": "Manage preferences",
  "banner_enabled": "always_enabled",
  "privacy_policy_link_label": "Privacy policy",
  "privacy_policy_url": "example.com/privacy",
  "privacy_preferences_link_label": "Manage preferences",
  "save_button_label": "Save",
  "is_default": true
}
```
* Visit the privacy experience tab and edit your configs

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [x] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [x] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

image

image

image

### CODE REVIEW

File: clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        isRequired
+      />
+      <CustomTextInput
+        label="Reject button displayed on the Banner and “Privacy preferences”"
+        name="reject_button_label"
+        variant="stacked"
+        isRequired
+      />
+      <CustomTextInput
+        label="“Privacy preferences” link label"
```
Comment (allisonking) [inline clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx:56]: Should this one be required as well? It looked like it's not required on the backend

**Comment by pattisdr** [comment]
Should this description be a little more generic for privacy center experience configs?
Screenshot 2023-05-31 at 10 55 31 AM

File: clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx (hunk 1 above)
Comment (pattisdr) [reply clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx:56]: Oh wait it is actually required for overlays not privacy centers, is this something required for both you think?

**Review by pattisdr — COMMENTED** [review]
Thank you for your quick work here!

This looks good, main thing I wanted to check here was that regions weren't removed even though they're missing from the form, but it looks like you're patching the entire request body with it's updates, so the original regions are getting patched back which works too.

File: clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx (hunk 1 above)
Comment (allisonking) [reply clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx:56]: ah I see, yeah I think it makes sense to be required for the overlay but not for the privacy center. I can make that change in the FE, and then the BE can stay the same?

Comment (pattisdr) [reply clients/admin-ui/src/features/privacy-experience/form/OverlayForm.tsx:56]: sounds good!

**Comment by allisonking** [comment]
> Thank you for your quick work here!
> 
> This looks good, main thing I wanted to check here was that regions weren't removed even though they're missing from the form, but it looks like you're patching the entire request body with it's updates, so the original regions are getting patched back which works too.

Ah yes, I now see you called that out in your PR—I added a test to make sure we aren't passing in empty regions if we didn't start with empty regions (as you said, we're passing in all of the original obj + modified fields) https://github.com/ethyca/fides/pull/3402/commits/7c413718d50f07e5284274bd7ca7699a812fd56b

**Review by pattisdr — APPROVED** [review]

**Pull request merged**